### ISSUE TITLE
git hooks are doubly broken on Windows

### ISSUE THREAD

**Issue by Rummskartoffel** [tridactyl/tridactyl#3022]
1. The dev dependency "shared-git-hooks" causes `yarn install` to fail because it receives a permission error when trying to create its symlinks. This can be prevented by running as admin, but we might as well just achieve the same effect by putting a `preinstall` script into package.json (`git config core.hooksPath hooks` does the trick) instead of relying on a partially broken npm package that hasn't been updated in ~2 years. See also: kilianc/shared-git-hooks#8
1. `post-checkout` seems to work fine (or at least it doesn't report any errors), but `pre-commit` creates a file called "nul" (which, btw, is a reserved file name on Windows that you can't normally create – or delete) that contains a bunch of linter errors, whenever I try to commit changes (to .ts files, at least). The first time I made a commit to (a fork of) this repository, *something* also created two weirdly named directories ("lint.*&lt;random letters&gt;*", IIRC), but I don't remember whether that was the hook or `yarn run pretty`, which I dutifully ran after the hook told me to.

**Comment by bovine3dom** [comment]
We should definitely investigate why 1) doesn't happen on CI. Half the point of it was to stop us breaking building on Windows.

**Comment by Rummskartoffel** [comment]
> 
> 
> We should definitely investigate why 1) doesn't happen on CI. Half the point of it was to stop us breaking building on Windows.
As far as I could find out, that is because GitHub Actions always run as admin, so they can create those symlinks just fine.

**Comment by Rummskartoffel** [comment]
I've started looking into this a little more, and it turns out "nul" is basically the MS-DOS version of /dev/null, so I guess Git Bash tried to translate the write to /dev/null done by eslint in the pre-commit hook and wrote to a normal file instead of the device file for some reason. The strangely-named directories I mentioned apparently came from that hook as well, as they are *supposed* to be created in .temp/, but for some reason they just... weren't.
I'm also gonna go and see if I can't make these hooks work in Git Bash/MinGW, because it turns none of them worked after all, which I somehow hadn't noticed until today, and I think the convoluted shared-git-hooks thing is pretty unnecessary.

**Comment by Rummskartoffel** [comment]
Another update: making the hooks *run* was not an issue, but making them *work* could prove more difficult. I've found out the following things:
* under MinGW, node's `fs.writeFileSync("/dev/null", "foo")` works as expected, and `fs.writeFileSync("nul", "foo")` creates a normal file called "nul", arguably also expected.
* under cmd.exe, node throws an error trying to write to  "/dev/null" as expected, but it writes to "nul" as a normal file instead of to the device file – but that file otherwise still behaves like it should on Windows, e.g. being impossible to delete through the file explorer. 
* here's where it gets weird: *something* appears to be translating the actual *invocation* of eslint that happens in `eslintUgly()`. https://github.com/tridactyl/tridactyl/blob/45bfe98ada7d69881cbc6e009bee1f92be000053/scripts/common.sh#L43 It's more easily observable when using yarn to run eslint manually, because yarn repeats back the command it is going to run,
    ```
    $ yarn run eslint -o /dev/null src/excmds.ts
    yarn run v1.22.10
    $ F:\projects\tridactyl\node_modules\.bin\eslint -o nul src/excmds.ts
    ```
    but it also happens when running the same command directly, as evidenced by "nul" being created.

I'm not sure how this happens or what to do about it. It can't be caused by yarn, because it happens without it as well, same with cmd.exe. It can't be eslint or node, because it happens before they are even invoked. That *would* leave only MinGW, but at the moment I can't think of any way of finding out for sure, let alone fixing it, or any reason this even happens in the first place.

**Comment by bovine3dom** [comment]
Let's just skip the pre-commit hooks on Windows : )

**Comment by Rummskartoffel** [comment]
Yeah, that's probably easiest. I'm gonna do a bit of testing to make sure that at least works and then I'll submit a PR. After all, I can always can come back to this in case I think of something.

**Comment by bovine3dom** [comment]
Fixed in #3033
